File: leetcode/2914. Minimum Number of Changes to Make Binary String Beautiful/main.go
package main

// 10
// 1001 -> 10 | 01
// 100101 -> 10 | 01 | 01

func minChanges(s string) int {
	result := 0

	for i := 0; i <= len(s)-2; i += 2 {
		substr := s[i : i+2]

		if substr == "01" || substr == "10" {
			result++
		}
	}

	return result
}

File: leetcode/2914. Minimum Number of Changes to Make Binary String Beautiful/main.ts
const minChanges = (s: string): number => {
  let result = 0;

  for (let i = 0; i <= s.length - 2; i += 2) {
    let part = s.substring(i, i + 2);

    if (part == "01" || part == "10") {
      result++;
    }
  }

  return result;
};
